File: opamenu-commons/OpaMenu.Infrastructure.Shared/Entities/Opamenu/TabEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using OpaMenu.Infrastructure.Shared.Enums.Opamenu;

namespace OpaMenu.Infrastructure.Shared.Entities.Opamenu;

/// <summary>
/// Tabela de comandas
/// </summary>
[Table("tabs")]
public class TabEntity : BaseEntity
{
    [Required]
    [Column("table_id")]
    public Guid TableId { get; set; }

    [ForeignKey(nameof(TableId))]
    public virtual TableEntity Table { get; set; } = null!;

    [MaxLength(50)]
    [Column("name")]
    public string? Name { get; set; }

    [Column("status")]
    public ETabStatus Status { get; set; } = ETabStatus.Open;

    [Column("opened_at")]
    public DateTime OpenedAt { get; set; } = DateTime.UtcNow;

    [Column("closed_at")]
    public DateTime? ClosedAt { get; set; }

    public virtual ICollection<OrderEntity> Orders { get; set; } = new List<OrderEntity>();
}

File: opamenu-garcom-mobile/lib/src/features/tabs/domain/entities/tab_entity.dart
class TabEntity {
  final String id;
  final String tableId;
  final String? name;
  final int status;
  final DateTime openedAt;
  final DateTime? closedAt;

  const TabEntity({
    required this.id,
    required this.tableId,
    required this.name,
    required this.status,
    required this.openedAt,
    required this.closedAt,
  });

  bool get isOpen => status == 1;
}

